api/app/files/api.py

```diff
@@ -45,9 +46,33 @@ def get(self):
             return send_file(temp.name, as_attachment=True, attachment_filename=args['filename'], mimetype='application/pdf')
 
 
+
+class FileAPIImproved(FileUploadMixin, restful.Resource):
+    def get(self, file_id):
+        # TO DO: Query the File table
+
+        req_parser = reqparse.RequestParser()
+        req_parser.add_argument('filename', type=str, required=True)
+        args = req_parser.parse_args()
+        LOGGER.debug("FileUpload GET args: {}".format(args))
+

```

Comment:
Then to retrieve the file object, look at some of the other repository files in the project and implement a `get_by_id `method. And then use that method to retrieve the relevant file object. There's plenty of examples in the project to see how to do this. If you're still struggling let me know.